### PR TITLE
Create an internal undeprecated function to check checkpoint exists.

### PR DESCRIPTION
**Pull request by abdullahselek**
Related issue [#30119](https://github.com/tensorflow/tensorflow/issues/30119l) for this MR.

### CODE REVIEW

File: tensorflow/python/training/saver.py [hunk 1]
```diff
@@ -1276,7 +1276,7 @@ def restore(self, sess, save_path):
     if save_path is None:
       raise ValueError("Can't load save_path when it is None.")
 
-    if not checkpoint_management.checkpoint_exists(compat.as_text(save_path)):
+    if not os.path.exists(compat.as_text(save_path)):
```
Comment (allenlavoie) [inline tensorflow/python/training/saver.py]: I'd use [gfile.Exists](https://github.com/tensorflow/tensorflow/blob/d23b8115e5db07805306ebbbf08c588b86e9eb27/tensorflow/python/platform/gfile.py#L26) (gfile is already imported in this file). That'll work with file systems registered with TensorFlow. You'd also need to append ".index" for v2 checkpoints, and we probably still need to support v1 checkpoints...

If you just want to get rid of the deprecation warning, it's probably easier to make a non-deprecated internal-to-TF version and use that here (you can also call it from the deprecated public endpoint).

Comment (abdullahselek) [reply tensorflow/python/training/saver.py]: Thanks @allenlavoie for your suggestion, I just made some changes on `checkpoint_management.py` as you have mentioned.

File: tensorflow/python/training/checkpoint_management.py [hunk 2]
```diff
@@ -347,15 +347,11 @@ def latest_checkpoint(checkpoint_dir, latest_filename=None):
   return None
 
 
-@deprecation.deprecated(
-    date=None,
-    instructions="Use standard file APIs to check for files with this prefix.")
-@tf_export(v1=["train.checkpoint_exists"])
-def checkpoint_exists(checkpoint_prefix):
+def cp_exists(checkpoint_prefix):
```
Comment (allenlavoie) [inline tensorflow/python/training/checkpoint_management.py]: How about checkpoint_exists_internal; cp is pretty ambiguous.

Otherwise looks great.

Comment (abdullahselek) [reply tensorflow/python/training/checkpoint_management.py]: Just renamed the function.

File: tensorflow/python/training/checkpoint_management.py [hunk 3]
```diff
@@ -375,6 +371,27 @@ def checkpoint_exists(checkpoint_prefix):
...
+
+  Args:
+    checkpoint_prefix: the prefix of a V1 or V2 checkpoint, with V2 taking
+      priority.  Typically the result of `Saver.save()` or that of
+      `tf.train.latest_checkpoint()`, regardless of sharded/non-sharded or
+      V1/V2.
+  Returns:
+    A bool, true iff a checkpoint referred to by `checkpoint_prefix` exists.
+  """
+  return cp_exists(checkpoint_exists)
```
Comment (allenlavoie) [inline tensorflow/python/training/checkpoint_management.py]: Needs to follow the rename

Comment (abdullahselek) [reply tensorflow/python/training/checkpoint_management.py]: Sorry, fixed as well.

**Review by allenlavoie — DISMISSED** [review]
Great, thanks for fixing this!

**Comment by abdullahselek** [comment]
Could you please tell me what are the required changes?

**Review by allenlavoie — DISMISSED** [review]

**Comment by gbaned** [comment]
@abdullahselek Can you please check build failures? Thanks!

**Comment by abdullahselek** [comment]
I hope should be fine now.

**Review by allenlavoie — DISMISSED** [review]

**Comment by gbaned** [comment]
@abdullahselek Can you please check ubuntu sanity errors? Thanks!

**Comment by abdullahselek** [comment]
Sanity errors should have gone with last changes.

**Comment by abdullahselek** [comment]
@allenlavoie or @gbaned could you please check my last changes?

**Review by allenlavoie — DISMISSED** [review]

**Comment by abdullahselek** [comment]
Rebased from original repo and repushed, please check @allenlavoie.

**Review by allenlavoie — APPROVED** [review]

**Comment by abdullahselek** [comment]
@allenlavoie I guess failing builds are not related to my changes, what is your suggestion?

**Comment by allenlavoie** [comment]
No worries then, it'll get pulled.

**Pull request merged**